services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSpec.groovy

```diff
@@ -413,58 +456,40 @@ class SwitchValidationSpec extends HealthCheckSpecification {
         flowHelper.addFlow(flow)
 
         when: "Delete created rules on the srcSwitch"
-        def createdCookies = northbound.getSwitchRules(switchPair.src.dpId).flowEntries.findAll {
-            !Cookie.isDefaultRule(it.cookie)
-        }*.cookie
-        assert createdCookies.size() == 2
-
-        northbound.deleteSwitchRules(switchPair.src.dpId, DeleteRulesAction.IGNORE_DEFAULTS)
+        def egressCookie = database.getFlow(flow.id).reversePath.cookie.value
+        northbound.deleteSwitchRules(switchPair.src.dpId, egressCookie)
 
         then: "Rule info is moved into the 'missing' section on the srcSwitch"
-        def srcSwitchValidateInfo = northbound.validateSwitch(switchPair.src.dpId)
-        srcSwitchValidateInfo.rules.missing.containsAll(createdCookies)
-        switchHelper.verifyRuleSectionsAreEmpty(srcSwitchValidateInfo, ["proper", "excess"])
+        verifyAll(northbound.validateSwitch(switchPair.src.dpId)) {
+            it.rules.missing == [egressCookie]
+            it.rules.proper.findAll {
+                !Cookie.isDefaultRule(it)
+            }.size() == 1 //ingress rule is left untouched
+            it.rules.excess.empty
+        }
 
         and: "Rule info is NOT moved into the 'missing' section on the dstSwitch and transit switches"
         def dstSwitchValidateInfo = northbound.validateSwitch(switchPair.dst.dpId)
-        dstSwitchValidateInfo.rules.proper.containsAll(createdCookies)
+        dstSwitchValidateInfo.rules.proper.findAll { !Cookie.isDefaultRule(it) }.size() == 2
         switchHelper.verifyRuleSectionsAreEmpty(dstSwitchValidateInfo, ["missing", "excess"])
         def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id)*.dpId
         def transitSwitches = involvedSwitches[1..-2].findAll { !it.description.contains("OF_12") }
 
         transitSwitches.each { switchId ->
             def transitSwitchValidateInfo = northbound.validateSwitch(switchId)
-            assert transitSwitchValidateInfo.rules.proper.containsAll(createdCookies)
             assert transitSwitchValidateInfo.rules.proper.findAll { !Cookie.isDefaultRule(it) }.size() == 2
             switchHelper.verifyRuleSectionsAreEmpty(transitSwitchValidateInfo, ["missing", "excess"])
         }
 
-        when: "Delete rules on the dstSwitch and transit switches"
-        involvedSwitches[1..-1].each { switchId ->
-            northbound.deleteSwitchRules(switchId, DeleteRulesAction.IGNORE_DEFAULTS)
+        when: "Synchronize the switch"
+        verifyAll(northbound.synchronizeSwitch(switchPair.src.dpId, false)) {
```

Comment:
`northbound.synchronizeSwitch(switchPair.src.dpId, false).rules.installed == [egressCookie]`

Reply:
done both

Reply:
this is a mistake. `with` and `verifyAll` are forcing the verification in 'where' block. So it was correct. Your option is missing an `assert` keyword

Reply:
All related comments are reverted to this style:
```
        when: "Synchronize the switch"
        with(northbound.synchronizeSwitch(switchPair.src.dpId, false)) {
            rules.installed == [egressCookie]
        }
```